packages/vats/test/bootstrapTests/bench-vaults-performance.js

```diff
@@ -159,15 +167,47 @@ test.skip('stress vaults', async t => {
   };
 
   // clear out for a baseline
-  await snapshotHeap('start');
+  await collectStats('start', dumpHeap);
   // 10 is enough to compare retention in heaps
   await openN(10, 1);
-  await snapshotHeap('round1');
+  await collectStats('round1', dumpHeap);
   await openN(10, 2);
-  await snapshotHeap('round2');
+  const memStats = await collectStats('round2', dumpHeap);
 
   // let perfObserver get the last measurement
   await eventLoopIteration();
 
+  const name = `stress-vaults${dumpTag}`;
+  let where = import.meta.url;
+  const idx = where.lastIndexOf('agoric-sdk/');
+  if (idx > 0) {
+    where = where.substring(idx);
+  }
+  const benchmarkReport = {
+    ...rows[1],
+    memStats,
+    name,
+    test: t.title,
+    where,
+  };
+  fs.writeFileSync(
+    `benchmark-${name}.json`,
+    JSON.stringify(benchmarkReport, null, 2),
+  );
+
   console.table(rows);
+}
+
+// Note: it is probably not useful to enable both of the two following benchmark
+// tests at the same time.  Nothing bad per se will happen if you do, but it
+// will take longer to run with no particular benefit resulting.  However, if you run
+// both you *must* run them serially, so that their executions don't get
+// comingled and mess up the numbers.
+
+test.skip('stress vaults with heap snapshots', async t => {
+  await stressVaults(t, true);
+});
+
+test.serial('stress vaults', async t => {
+  await stressVaults(t, false);
```

Comment:
if you make the flag optional,
```suggestion
  await stressVaults(t);
```

Reply:
I like the symmetry in the code here. Make it more visual in the difference between two.

Reply:
I think that's a significant uglification.  This is not a function that's going to be called several different ways from several different places, it's going to be called in exactly these two ways from exactly these two places.  The two parallel uses of `stressVaults` that are presented in immediate succession, one each for the `true` and `false` values of the dump flag.  I like the fact that the explicit parameter underscores the symmetry of the two.

Reply:
convinced!